src/pages/[post].tsx

```diff
@@ -428,13 +384,25 @@ export default function PostPage({
           ],
         }}
       />
+
       {videoResource && (
-        <PostPlayer
-          playbackId={videoResource.fields.muxPlaybackId}
-          eggheadLessonId={post.fields.eggheadLessonId}
-          post={post}
-        />
+        <div>
+          <MuxPlayerProvider>
+            <VideoPlayerOverlayProvider>
+              <div className="relative h-full w-full">
+                <VideoPlayerOverlay resource={post} />
+                <PostPlayer
+                  playbackId={videoResource.fields.muxPlaybackId}
+                  eggheadLessonId={post.fields.eggheadLessonId}
+                  post={post}
+                  postTags={tags}
+                />
+              </div>
+            </VideoPlayerOverlayProvider>
+          </MuxPlayerProvider>
```

Comment:
adding mux and overlay provider which gives controls for overlay to be triggered and access mux player to rewatch lesson